woocommerce/woocommerce
# Improve remote logging structure and content

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/51355


Currently, we're logging the `extra` field in our error logs with a structure like this:

```json
{
  "source": "fatal-errors",
  "error": {
    "type": 4,
    "file": "\/srv\/htdocs\/wp-content\/affegg-parsers\/MercadolibrecommxParser.php",
    "line": 38
  }
}
```

This structure has two main issues:

1. The `file` field contains the server logstash file path, which is not very useful for debugging purposes.
2. The `error` information is nested within the `extra` field, making it less accessible.
 
Propose the following improvements:

1. Move the `file` information directly to the `file` field in the log structure.
2. Remove the `error` object from the context and flatten the structure.

The proposed new structure would look like this:

```json
{
  "source": "fatal-errors",
  "file": "wp-content\/affegg-parsers\/MercadolibrecommxParser.php",
  "error": {
    "type": 4,
    "line": 38
  }
}
```



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

To test this PR on JN, we need to merge and deploy D161391-code first. Otherwise, you can test this PR on your local environment by using sandbox with that patch applied.

1. Create a fresh site
2. Go to Tools -> WCA Test Helper -> Remote Logging
3. Enable remote logging
4. Click on PHP Simulate Core Exception, refresh the page and confirm it throws an error
5. See PCYsg-11XN-p2 for instructions on viewing the logs
6. Confirm the file field is now `**/woocommerce-beta-tester/woocommerce-beta-tester.php:148`



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
